### ISSUE TITLE
Start VM as Headless

### ISSUE THREAD

**Issue by sandeep-khanna** [onlyfang/VBoxVmService#16]
My goal is to have a Linux virtual machine up and available at all times when logged in to my account on the host. The guest machine should also survive reboots and shutdowns. 

**Host:** Microsoft Windows 10 Enterprise
**Guest:** Ubuntu 18.04

Two solutions that seem promising:

**1. [VBoxHeadlessTray](https://github.com/toptensoftware/VBoxHeadlessTray)**

- Great concept
- Nice tray icon that is animated when starting/restoring a VM
- Starts guest VM is headless mode so the VirtualBox GUI can be used to just 'Show' the running VM
- Not very reliable; Guest VM looses saved state possibly due to the VBoxHeadlessTray process being killed too soon by Windows during restart/shutdown.

**2. [VBoxVmService](https://github.com/onlyfang/VBoxVmService)**
- Interesting take and similar to the VBoxHeadlessTray
- Default install location (C:\vms) should instead be "C:\Program Files\Oracle\VirtualBox\VBoxVmService"
- Tray service icon is uninteresting and limited in actions that can be performed
- Seems to start guest VMs (atleast based on the logs); Missing ability for using VirtualBox manager GUI to 'Show' the running VM.
- Documentation could be better

What is the best or recommended approach to 'connecting' or 'showing' a VBoxVmService started virtual machine?

**Comment by caffeine74** [comment]
I expect RDP or set up SSH and log in securely.

**Comment by bkraul** [comment]
Yeah but what if the machine's networking is hosed and the only way you have is through Vbox console? With VBoxVmService you can't do this, as the GUI will not see the current vm process...

**Comment by BOBofBORG** [comment]
I know this is old but wanted to provide this feedback.  Search for "headless virtualbox" and you should come up with several instructions and wiki's.  I have this setup on 2 servers:  windows 2012 & linux server.  For the windows server, i use vboxvmservice & phpvirtualbox.  This would allow you to connect remotely to start or stop the vm's thru a website.  

If you have network issues with the VM's, you can also rdp into the vm using the host server ip & port of the VM.

**Comment by bkraul** [comment]
Yeah, tried all that. I eventually just settled with using in-session headless vbox and gave up on the service approach. Seems to work well. It's sad that virtual box does not support a robust session-less service setup out of the box. VMWare has this useful feature.
